client/src/core/props-behavior.js

```diff
@@ -131,107 +131,148 @@ class PropsSelector {
   }
 
   /**
-   * Select a prop for building mode
-   * @param {Vector2} pointer - 2D Pointer for the raycaster.
-   * @param {boolean} add - True to add the newfound prop to the existing
-   *                        selection list, false (default) to clear the
-   *                        list first.
+   * Traverse up the object hierarchy to find the root prop object.
+   *
+   * @param {Object3D} object - The child object to start from.
+   * @returns {Object3D} The root object containing userData.prop,
+   *                           or the original object if none found.
    */
-  select(pointer, add = false) {
-    let done = false;
+  getRootPropObject(object) {
+    while (object && !object.userData?.prop && object.parent) {
+      object = object.parent;
+    }
+    return object;
+  }
 
-    this.clickRaycaster.setFromCamera(
-        pointer, this.engine3d.camera,
-    );
+  /**
+  * Add a single root prop to the current selection list.
+  *
+  * @param {Object3D} rootProp - Root object representing the prop.
+  */
+  selectSingleProp(rootProp) {
+    const boundingBox = rootProp.getObjectByName(boundingBoxName)?.clone();
+    if (!boundingBox) return;
+
+    const {x, y, z} = rootProp.userData.prop;
+
+    const stagingProp = rootProp.clone();
+    stagingProp.visible = true;
+    stagingProp.position.set(x, y, z);
+
+    this.engine3d.appendToNode(this.stagingNode, stagingProp);
+    stagingProp.updateMatrix();
+
+    boundingBox.position.set(x, y, z);
+    boundingBox.rotation.copy(rootProp.rotation);
+    boundingBox.updateMatrix();
+    boundingBox.visible = true;
+    this.engine3d.addHelperObject(boundingBox);
+
+    this.props.push({prop: rootProp, stagingProp, boundingBox});
+  }
 
-    const intersects =
-        this.clickRaycaster.intersectObjects(
-            this.engine3d.getSelectableNodes(), true,
+  /**
+  * Select all props in the same chunk within a given radius.
+  *
+  * @param {Object3D} rootProp - Root prop used as center point.
+  * @param {number} radius - Distance (in meters) to include nearby props.
+  */
+  selectNearbyProps(baseProp) {
+    const chunkHandle = baseProp.userData.chunkNodeHandle;
+    if (!chunkHandle) return;
+
+    const chunkId = this.worldManager.chunkKeys.get(chunkHandle);
+    if (!chunkId) return;
+
+    // Find all props in the same chunk
+    // eslint-disable-next-line
+    for (const [propId, obj3d] of this.worldManager.props.entries()) {
```

Comment:
This looks _very_ inefficient the more props there are, given that you already have the chunkNodeHandle (which is just an engine3d node ID), you can add something to the Engine3D class itself to return the children of a given node (which will be the props within the chunk here), I'm even surprised we don't have this yet but I guess we just didn't need it until now.

You can look into `belongsToNode` to get an idea, you can add your own `getNodeContent(...)` method to return said children from the node.